cypress/integration/plugins/anomaly-detection-dashboards-plugin/historical_analysis_spec.js

```diff
@@ -44,16 +46,43 @@ describe('Historical results page', () => {
           expect(parseInt(anomalyOccurrenceCount)).to.equal(0);
         });
     });
+    cy.getElementByTestId('anomalyOccurrenceTab').click();
     cy.getElementByTestId('anomalyOccurrencesHeader').should('contain', '(0)');
   };
 
   // Creating a sample detector and visiting the config page
   before(() => {
     cy.visit(AD_URL.OVERVIEW);
-    cy.getElementByTestId('createHttpSampleDetectorButton').click();
-    cy.visit(AD_URL.OVERVIEW);
-    cy.getElementByTestId('viewSampleDetectorLink').click();
-    cy.getElementByTestId('historicalTab').click();
+    cy.get('[data-test-subj=createHttpSampleDetectorButton]').then(($btn) => {
+      if ($btn.is(':disabled')) {
+        cy.getElementByTestId('viewSampleDetectorLink').click();
+        cy.getElementByTestId('configurationsTab').click();
+        cy.getElementByTestId('detectorIdCell').within(() => {
+        cy.get('.euiText--medium')
+          .invoke('text')
+          .then((detectorId) => {
+            cy.log('Stopping detector with ID: ' + detectorId);
+            cy.stopDetector(detectorId);
+            cy.wait(10000);
+            cy.log('Deleting detector with ID: ' + detectorId);
+            cy.deleteDetector(detectorId);
+            cy.log('Deleting index with name: ' + indexName);
+            cy.deleteIndex(indexName);
+          });
+          cy.wait(10000);
+          cy.visit(AD_URL.OVERVIEW);
```

Comment:
Thanks for adding this workaround! I think a long-term cleaner soln will be some util fns to bulk cleanup any existing detectors by deleting the system indices and any created sample indices, that we can run before and/or after each test suite.